sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -1698,6 +1739,18 @@ private Map<String, String> getRequestHeaders(RequestOptions options, ResourceTy
             headers.put(HttpConstants.HttpHeaders.IF_NONE_MATCH, options.getIfNoneMatchETag());
         }
 
+        if (options.getReadConsistencyStrategy() != null
```

Comment:
Looks like repeat logic?

Reply:
Not really - one based off of client level setting, one request level